cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -136,6 +138,33 @@ class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
       fs::FileLocator destination_locator) const override;
 
   std::shared_ptr<FileWriteOptions> DefaultWriteOptions() override;
+   
+  /// \brief A getter function to retrieve the dataset encryption configuration
+  std::shared_ptr<parquet::encryption::DatasetEncryptionConfiguration> GetDatasetEncryptionConfig() const {
+    return dataset_encryption_config_;
+  }
+  /// \brief A getter function to retrieve the dataset decryption configuration
+  std::shared_ptr<parquet::encryption::DatasetDecryptionConfiguration> GetDatasetDecryptionConfig() const {
+    return dataset_decryption_config_;
+  }
```

Comment:
Instead of putting these properties on `ParquetFileFormat` can you put the decryption config on `ParquetFragmentScanOptions` and the encryption config on `ParquetFileWriteOptions`?

Reply:
I wouldn't want to add new pyarrow properties as part of the 12.0.0 release that we then have to do a deprecation cycle for.  However, it seems like the user-facing python API wouldn't have to change if we changed this internally.  If you agree then I think this can be a follow-up.

Reply:
I've filed #35211 for the follow-up.